pkg/core/template/engine.go

```diff
@@ -1,24 +1,32 @@
 package template
 
 import (
+	"sort"
+
+	"golang.org/x/text/language"
+
 	"github.com/skygeario/skygear-server/pkg/core/config"
 )
 
 // Engine resolves and renders templates.
 type Engine struct {
-	DefaultLoader *StringLoader
-	URILoader     *URILoader
+	DefaultLoader         *StringLoader
+	URILoader             *URILoader
+	TemplateItems         []config.TemplateItem
+	PreferredLanguageTags []string
 }
 
 type RenderOptions struct {
 	Required bool
 	Key      string
 }
 
-func NewEngine(enabledFileLoader bool) *Engine {
+func NewEngine(enabledFileLoader bool, templateItems []config.TemplateItem, tags []string) *Engine {
```

Comment:
When we develop user preferred languages in the future, most likely we will need to get the language tags from the auth context.
Maybe we should add some comment to describe why we want to put the preferred languages tags here, we may need to update the middleware in future and see how can we get the preferred language in the DI function.